### PR TITLE
[ci-visibility] Change `test.type` in browser tests 

### PR DESCRIPTION
**Pull request by ericlaz**
### What does this PR do?
Adds a new test type: `browser` which is only enabled when testing using Cypress or Playwright.

### Motivation
It adds more flexibility to test types

### Plugin Checklist

- [X] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/plugins/util/ci.js [hunk 1]
```diff
@@ -81,6 +81,13 @@ function filterSensitiveInfoFromRepository (repositoryUrl) {
   }
 }
 
+function getTestType (frameworkName) {
+  if (frameworkName === 'playwright' || frameworkName === 'cypress') {
+    return 'browser'
+  }
+  return 'test'
+}
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/util/ci.js:89]: Just to get a bit more consistency on the naming
```suggestion
function getTestTypeFromFramework (testFramework) {
  if (testFramework === 'playwright' || testFramework === 'cypress') {
    return 'browser'
  }
  return 'test'
}
```

File: packages/datadog-plugin-cypress/src/plugin.js [hunk 2]
```diff
@@ -37,7 +40,7 @@ const CYPRESS_STATUS_TO_TEST_STATUS = {
 function getTestSpanMetadata (tracer, testName, testSuite, cypressConfig) {
   const childOf = getTestParentSpan(tracer)
 
-  const commonTags = getTestCommonTags(testName, testSuite, cypressConfig.version)
+  const commonTags = getTestCommonTags(testName, testSuite, cypressConfig.version, getTestType(TEST_FRAMEWORK_NAME))
```
Comment (juan-fernandez) [inline packages/datadog-plugin-cypress/src/plugin.js:43]: I think we can avoid importing a new function and just letting that responsibility to `getTestCommonTags`
```suggestion
  const commonTags = getTestCommonTags(testName, testSuite, cypressConfig.version, TEST_FRAMEWORK_NAME)
```
(see comments below)

File: packages/dd-trace/src/plugins/util/test.js [hunk 3]
```diff
@@ -188,10 +188,10 @@ function getTestParentSpan (tracer) {
   })
 }
 
-function getTestCommonTags (name, suite, version) {
+function getTestCommonTags (name, suite, version, testType = 'test') {
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/util/test.js:191]:
```suggestion
function getTestCommonTags (name, suite, version, testFramework) {
```

File: packages/dd-trace/src/plugins/util/test.js [hunk 4]
```diff
@@ -188,10 +188,10 @@ function getTestParentSpan (tracer) {
...
   return {
     [SPAN_TYPE]: 'test',
-    [TEST_TYPE]: 'test',
+    [TEST_TYPE]: testType,
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/util/test.js:194]:
```suggestion
    [TEST_TYPE]: getTestType(testFramework),
```

File: packages/dd-trace/src/plugins/util/ci.js (hunk 1 above)
Comment (juan-fernandez) [reply packages/dd-trace/src/plugins/util/ci.js:89]: we can define `getTestTypeFromFramework` within `packages/dd-trace/src/plugins/util/test.js` and it wouldn't need to be exported

**Review by juan-fernandez — COMMENTED** [review]
good job! Most of the comments are just for moving certain things around, but the logic is good

File: packages/dd-trace/src/plugins/ci_plugin.js [hunk 5]
```diff
@@ -111,7 +111,10 @@ module.exports = class CiPlugin extends Plugin {
     const childOf = getTestParentSpan(this.tracer)
 
     let testTags = {
-      ...getTestCommonTags(testName, testSuite, this.frameworkVersion),
+      ...getTestCommonTags(testName,
+        testSuite,
+        this.frameworkVersion,
+        this.constructor.id),
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/ci_plugin.js:117]: we don't force any style, but this is more in line with the rest of the code
```suggestion
      ...getTestCommonTags(
        testName,
        testSuite,
        this.frameworkVersion,
        this.constructor.id
      ),
```

File: packages/dd-trace/src/plugins/util/ci.js [hunk 6]
```diff
@@ -81,6 +81,13 @@ function filterSensitiveInfoFromRepository (repositoryUrl) {
   }
 }
 
+function getTestTypeFromFramework (testFramework) {
+  if (testFramework === 'playwright' || testFramework === 'cypress') {
+    return 'browser'
+  }
+  return 'test'
+}
+
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/util/ci.js:90]: let's move this function to `packages/dd-trace/src/plugins/util/test.js` so we don't have to export/import it

**Review by juan-fernandez — COMMENTED** [review]
we're almost there :-) 

additionally, cypress tests are failing because we need to update this https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-plugin-cypress/test/index.spec.js#L81 and this https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-plugin-cypress/test/index.spec.js#L106


we could also optionally expand this test `integration-tests/playwright/playwright.spec.js` to include `TEST_TYPE` assertion, but that's up to you 😄

File: packages/dd-trace/src/plugins/util/test.js [hunk 7]
```diff
@@ -173,6 +173,13 @@ function getTestParametersString (parametersByTestName, testName) {
   }
 }
 
+function getTestTypeFromFramework (testFramework) {
+  if (testFramework.includes('playwright') || testFramework.includes('cypress')) {
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/util/test.js:177]: let's simply do. Those frameworks should be fixed
```suggestion
  if (testFramework === 'playwright' || testFramework === 'cypress') {
```

File: integration-tests/playwright/playwright.spec.js [hunk 8]
```diff
@@ -73,7 +73,7 @@ versions.forEach((version) => {
             assert.equal(testSessionEvent.content.meta[TEST_STATUS], 'fail')
             assert.include(testModuleEvent.content.resource, 'test_module.playwright test')
             assert.equal(testModuleEvent.content.meta[TEST_STATUS], 'fail')
-
+            assert.equal(testSessionEvent.content.meta[TEST_TYPE], 'browser')
```
Comment (juan-fernandez) [inline integration-tests/playwright/playwright.spec.js:76]: nice!

**Review by juan-fernandez — COMMENTED** [review]
I just noticed that we're missing a `TEST_TYPE` update in https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/util/test.js#L277. 

The tests are correctly set but session, modules and suites are still being reported with `test.type='test'`

**Comment by juan-fernandez** [comment]
> I just noticed that we're missing a `TEST_TYPE` update in https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/util/test.js#L277.
> 
> The tests are correctly set but session, modules and suites are still being reported with `test.type='test'`

that's why the integration test is failing: https://github.com/DataDog/dd-trace-js/actions/runs/5487302751/jobs/9998579728?pr=3358. Good thing we added that! 😄

**Review by rochdev — APPROVED** [review]

**Pull request merged**